lib/Frontend/FrontendOptions.cpp

```diff
@@ -73,11 +73,11 @@ bool FrontendInputs::verifyInputs(DiagnosticEngine &Diags, bool TreatAsSIL,
       Diags.diagnose(SourceLoc(), diag::error_repl_requires_no_input_files);
       return true;
     }
-  } else if (TreatAsSIL && hasPrimaryInput()) {
+  } else if (TreatAsSIL && havePrimaryInputs()) {
     // If we have the SIL as our primary input, we can waive the one file
     // requirement as long as all the other inputs are SIBs.
     for (unsigned i = 0, e = inputFilenameCount(); i != e; ++i) {
-      if (i == getPrimaryInput()->Index)
+      if (i == getOptionalPrimaryInput()->Index)
```

Comment:
You haven't checked that there's exactly one primary input here.

Reply:
Fixed.